### DISCUSSION TITLE
automatic shopping category for food child

### DISCUSSION THREAD

**Discussion by TWART016** [TandoorRecipes/recipes#1319] [Ideas]
I defined a parent `Cheese`. Sub-Ingredients should be a child of the parent. 

The parent have to shopping category `Milk`, the child has no category. If I move the Sub-Ingedients to the parent the child does not get the shopping category.

Does it make sense to automatic set the category to use it in shopping list?
![image](https://user-images.githubusercontent.com/29805669/149604635-8c207ef4-7b4e-4541-adad-1924116462a6.png)

**Comment by smilerz** [comment]
Children need to be set to inherit supermarket categories. You can set default inherited settings in Space settings

**Reply by TWART016** [reply]
Where can I find this setting?

[the remark this answers is not in the document]
**Reply by TWART016** [reply]
I can't find anything with inherit. There is everything read only.

**Comment by smilerz** [comment]
Are you an admin?

**Reply by TWART016** [reply]
Yes, I have all permissions.

**Comment by smilerz** [comment]
Which version are you running?

**Reply by TWART016** [reply]
v1.0.4

**Comment by smilerz** [comment]
Got it.  That setting is available next release.

**Comment by TWART016** [comment]
Now with 1.0.5.2 there is in Space Settings the food inherit available for `Supermarket Category` and `Ignore Shopping` but not for `Shopping Category`. 

What does `Ignore Shopping` mean?


After add an inherit object I click `save`. The selected inherit is not visible, only after a reload of the page. Same with deleting. Is this a bug?

**Comment by smilerz** [comment]
Supermarket category and shopping category are the same thing. 

Ignore shopping will always exclude an item from the shopping list. But isn’t enabled yet 

what do you mean by “the selected inherit isn’t visible yet”?

**Reply by TWART016** [reply]
My parent food has a category. Then I move one food to the parent as a child. Ths child still does not have the same category than the parent. When editing the child the inherit field is empty.
Even If I select the value in the value field the category is not set and in the Shopping List the category is Undefined.

If I enable ignore shopping is space settings all items will be ignored in the shopping list?

**Reply by smilerz** [reply]
> When editing the child the inherit field is empty.

This sounds like expected behavior.  The Food item you moved is not set to inherit any values.
At upgrade no food are set to inherit anything.  You can set default values to inherit in Space Settings - this will only change the behavior of new food.  To set all food to this default behavior select the 'Reset Food Inheritance'

> Even If I select the value in the value field the category is not set and in the Shopping List the category is Undefined.

To clarify - you are attempting to set a child shopping category.  The parent has a category defined, the child has no Inherit Fields set, but you are not allowed to save a shopping category to the child?

> If I enable ignore shopping is space settings all items will be ignored in the shopping list?

If you add Ignore Shopping to the inherit fields value, by default when foods are created they will inherit the Ignore Shopping attribute from their parent.

**Reply by TWART016** [reply]
I assigned a category to a food. Then I moved a food to another so their is parent and child. The child does not have the category even if in space settings `Supermarket Category` is enabled.
I thought If I move a food to a parent the category will be automatically set.

With new food (manually) I can set a category but not a parent.

What is the `Ignore Shopping` attribute? Add to shopping list and On Hold?

**Reply by smilerz** [reply]
> I assigned a category to a food. Then I moved a food to another so their is parent and child. The child does not have the category even if in space settings `Supermarket Category` is enabled. I thought If I move a food to a parent the category will be automatically set.
> 
Setting 'Supermarket Category' in space settings does not turn on inheritance for all foods - it only sets the default inheritance for new food.  To turn on inheritance for a single food you need to add 'supermarket category' as inherited on that food.  Or you can set all food to the default settings by selecting the 'Reset Food Inheritance' checkbox and clicking Save.

Foods that don't have a parent currently hide the inheritance, which is confusing.  I'm working on a different solution to hide inheritance if it isn't necessary.

> With new food (manually) I can set a category but not a parent.
>
I don't understand what this means.

> What is the Ignore Shopping attribute? Add to shopping list and On Hold?
>
Nothing yet.  It was overlooked when releasing.  I'm working on it now.  Once added you can mark a food 'Ignore Shopping' and it will never add it to the shopping list (e.g. water)

**Reply by TWART016** [reply]
Ok thanks. If I create or import a food it has no parent and therefore no category.

For now I change food to child and then click reset `Reset Food Inheritance` one time. 
It would be great if after the move the inheritance was automatically set without  `Reset Food Inheritance`

But I can wait, the workaround works.

**Reply by smilerz** [reply]
> Ok thanks. If I create or import a food it has no parent and therefore no category.
> 
> For now I change food to child and then click reset `Reset Food Inheritance` one time. It would be great if after the move the inheritance was automatically set without `Reset Food Inheritance`
> 
> But I can wait, the workaround works.

It should only be required the first time.  Every food now has those fields as inherited now and every new food will also have those items inherited.
